mobile-browser-based-version/src/helpers/training/training_manager.js

```diff
@@ -4,7 +4,43 @@ import {
   getWorkingModelMetadata,
 } from '../memory/helpers';
 
+import * as tf from '@tensorflow/tfjs';
+
+// Extra functions that are used to create Dataset object to be able to perform batch wise preprocessing of images.
+
+function trainDataGenerator(dataset, labels, trainingInformation) {
+  return function* dataGenerator() {
+    for (let i = 0; i < dataset.shape[0] * (1 - trainingInformation.validationSplit); i++) {
+      var tensor = tf.tensor(dataset.arraySync()[i]);
+      if (trainingInformation.resize) {
+        tensor = tf.image.resizeBilinear(
+          tensor,
+          [trainingInformation.RESIZED_IMAGE_H,
+          trainingInformation.RESIZED_IMAGE_W]);
+      }
+      yield { xs: tensor, ys: tf.tensor(labels.arraySync()[i]) }
+    }
+  }
+}
+
+function validationDataGenerator(dataset, labels, trainingInformation) {
```

Comment:
You can make a single helper function for both the train & validation datasets, which would be called for two different sets of indexes. The function name could be something more specific, as it returns an image dataset. Also, functions should not be in a class file but rather in a separate helper file :)

Technically you could go even further and pass a lambda such as `(tensor) => tf.image.resizeBilinear(tensor, [...]])` to the helper function, to generalize its purpose :)